### ISSUE TITLE
Can't find types with pnpm and moduleResolution=nodeNext

### ISSUE THREAD

**Issue by agriffis** [ariakit/ariakit#2356]
### Current behavior

I'm getting this error from `tsc`

```
src/quickstart/components/layout/Accordion/index.tsx:1:21 - error TS7016: Could not find a declaration file for module '@ariakit/react'. '/home/aron/src/ss/clients/tizra/evergreen/node_modules/.pnpm/@ariakit+react@0.1.6_react-dom@18.2.0_react@18.2.0/node_modules/@ariakit/react/cjs/index.cjs' implicitly has an 'any' type.
  Try `npm i --save-dev @types/ariakit__react` if it exists or add a new declaration (.d.ts) file containing `declare module '@ariakit/react';`

1 import * as AK from '@ariakit/react'
```

This was working for me with 2.0.0-next.43

### Steps to reproduce the bug

I couldn't repro this with codesandbox, maybe because it needs pnpm? Here's a minimal repo

https://github.com/agriffis/vite-ariakit-nodenext

```
git clone https://github.com/agriffis/vite-ariakit-nodenext
cd vite-ariakit-nodenext
pnpm i
pnpm tsc
```

### Expected behavior

tsc should be able to find the types

**Comment by agriffis** [comment]
I thought this was pnpm-specific but I just retried (with the same repo) using npm and yarn, and it reproduces with both. So it's purely the moduleResolution setting that's the problem.

**Comment by diegohaz** [comment]
Don't you have to set `"type": "module"` in your package.json if you're using `"moduleResolution": "nodenext"`?

**Comment by agriffis** [comment]
@diegohaz Sigh, if I do that then other things break. I've had it set this way for a while, and need it for react-query-devtools dynamic import to work. I think specifically `ts-node` scripts broke on me last time. Although it's been a while since I've tried, so I'll give it a go.

Thanks for the reply, I know this isn't really an ariakit problem specifically. Trying to get module and package resolution stuff all working together feels like a whack-a-mole game.

**Comment by agriffis** [comment]
I was able to switch back to `"moduleResolution": "node"` and everything is working now, including a number of other places that I had `// @ts-expect-error` because of seemingly missing types. Phewph! :sweat_smile:

**Comment by djMax** [comment]
I don't think this bug should be closed. I think it's an issue with the way the types are exported... Let me see if I can cobble together a PR.
